modules/gapi/test/gapi_gpu_test.cpp

```diff
@@ -202,6 +210,243 @@ TEST(GPU, Symm7x7_test)
+    const unsigned int refIntelVendorID = 0x8086; // Intel
+    IDXGIAdapter* out_adapter = nullptr;
+
+    while (adapter_factory->EnumAdapters(adapter_index, &out_adapter) != DXGI_ERROR_NOT_FOUND) {
+        DXGI_ADAPTER_DESC adapter_desc{};
+        out_adapter->GetDesc(&adapter_desc);
+        if (adapter_desc.VendorId == refIntelVendorID) {
+            intel_adapter = out_adapter;
+            break;
+        }
+        out_adapter->Release();
+        ++adapter_index;
+    }
+    if (!intel_adapter) {
+        throw SkipTestException("No Intel GPU adapter on aboard.");
+    }
+    intel_adapter->Release();
+
+    *device = nullptr;
+    hr = D3D11CreateDevice(nullptr,
+                           D3D_DRIVER_TYPE_HARDWARE,
+                           nullptr,
+                           0,
+                           nullptr,
+                           0,
+                           D3D11_SDK_VERSION,
+                           &(*device),
+                           nullptr,
+                           nullptr);
+
+    if(FAILED(hr)) {
+        CV_LOG_DEBUG(NULL, "D3D11CreateDevice was failed: " << std::to_string(HRESULT_CODE(hr)));
+        throw SkipTestException("D3D11CreateDevice was failed");
+    }
+    if (*device == nullptr) {
+        CV_LOG_DEBUG(NULL, "OpenCL: D3D11CreateDevice returns empty ID3D11Device\n");
+        throw SkipTestException("D3D11CreateDevice was failed");
+    }
+}
+
+void prepareD3D11Texture2D(const UINT width, const UINT height, std::vector<uint8_t>& test_data,
+                           ID3D11Texture2D** texture,
+                           ID3D11Device** device) {
+    // Create device
+    *device = nullptr;
+    createD3D11Device(device);
+
+    // Create texture description
+    D3D11_TEXTURE2D_DESC desc = {};
+    desc.Width = width;
+    desc.Height = height;
+    desc.MipLevels = 1;
+    desc.ArraySize = 1;
+    desc.Format = DXGI_FORMAT_A8_UNORM; // CV_8U
+    desc.SampleDesc.Count = 1;
+    desc.Usage = D3D11_USAGE_DEFAULT;
+    desc.BindFlags = D3D11_BIND_SHADER_RESOURCE;
+    desc.MiscFlags = D3D11_RESOURCE_MISC_SHARED;
+    desc.CPUAccessFlags = 0; // 0 if CPU access is not required
+
+    // Associate subresource with test data
+    std::iota(test_data.begin(), test_data.end(), uint8_t(0));
+    D3D11_SUBRESOURCE_DATA initData = { test_data.data(),
+                                        sizeof(uint8_t) * width,
+                                        height };
+
+    // Create texture2D by test data
+    HRESULT hr = (*device)->CreateTexture2D(&desc, &initData, &(*texture));
+
+    if(FAILED(hr)) {
+        CV_LOG_DEBUG(NULL, "CreateTexture2D was failed: " << std::to_string(HRESULT_CODE(hr)));
+        throw SkipTestException("CreateTexture2D was failed");
+    }
+
+    if(texture == nullptr) {
+        throw SkipTestException("CreateTexture2D was failed: texture ptr is NULL");
+    }
+}
+
+int findPlaceOfDevice(const std::vector<std::pair<bool, cv::ocl::Device>>& devices,
```

Comment:
by your function finding contract it is always true that  you do find prefereered devices before at first step, then you find compatible as the second step
That means you always have got preferred device as first element of vector. Moreover it is seem logically correct to keep the most preferrable device in the list beginning.
Thus there are no need to make such find place function but just check it:

devices.empy();
if (devices.begin()->first )  <--- it must be preferred device
otherwise it is compatible

Reply:
Removed. Use first element now.